packages/nimble-components/src/number-field/index.ts

```diff
@@ -40,11 +42,68 @@ export class NumberField extends mixinErrorPattern(
     @attr({ attribute: 'appearance-readonly', mode: 'boolean' })
     public appearanceReadOnly = false;
 
+    private decimalSeparator = '.';
+    private inputFilterRegExp = this.buildFilterRegExp(this.decimalSeparator);
+
+    private readonly langSubscriber: DesignTokenSubscriber<typeof lang> = {
+        handleChange: () => this.updateDecimalSeparatorAndInputFilter()
+    };
+
     public override connectedCallback(): void {
         super.connectedCallback();
 
         // This is a workaround for FAST issue: https://github.com/microsoft/fast/issues/6148
         this.control.setAttribute('role', 'spinbutton');
+        this.updateDecimalSeparatorAndInputFilter();
+        lang.subscribe(this.langSubscriber, this);
+    }
+
+    public override disconnectedCallback(): void {
+        super.disconnectedCallback();
+
+        lang.unsubscribe(this.langSubscriber, this);
+    }
+
+    protected override sanitizeInput(inputText: string): string {
+        return inputText.replace(this.inputFilterRegExp, '');
+    }
+
+    // this.value <-- this.control.value
+    protected override syncValueFromInnerControl(): void {
+        this.value = this.decimalSeparator !== '.'
+            ? this.control.value.replace(this.decimalSeparator, '.')
+            : this.control.value;
+    }
+
+    // this.value --> this.control.value
+    protected override syncValueToInnerControl(): void {
+        this.control.value = this.decimalSeparator !== '.'
+            ? this.value.replace('.', this.decimalSeparator)
+            : this.value;
+    }
+
+    private updateDecimalSeparatorAndInputFilter(): void {
+        const previousSeparator = this.decimalSeparator;
+        this.decimalSeparator = this.getSeparatorForLanguange(
+            lang.getValueFor(this)
+        );
+        this.inputFilterRegExp = this.buildFilterRegExp(this.decimalSeparator);
```

Comment:
This can move inside the `this.decimalSeparator !== previousSeparator` check now